src/libraries/System.Private.CoreLib/src/System/Runtime/Serialization/SerializationInfo.SerializationGuard.cs

```diff
@@ -64,14 +65,7 @@ public static void ThrowIfDeserializationInProgress()
         public static void ThrowIfDeserializationInProgress(string switchSuffix, ref int cachedValue)
         {
             const string SwitchPrefix = "Switch.System.Runtime.Serialization.SerializationGuard.";
-            if (switchSuffix == null)
-            {
-                throw new ArgumentNullException(nameof(switchSuffix));
-            }
-            if (string.IsNullOrWhiteSpace(switchSuffix))
-            {
-                throw new ArgumentException(SR.Argument_EmptyName, nameof(switchSuffix));
-            }
+            Debug.Assert(!string.IsNullOrEmpty(switchSuffix));
```

Comment:
```suggestion
            Debug.Assert(!string.IsNullOrWhiteSpace(switchSuffix));
```

(Keeps the original check intact.)